### PR TITLE
feat: adapt to new core

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -23,16 +23,15 @@ def docker_compose(request):
 @pytest.fixture()
 def get_documents():
     def get_documents_inner(nr=10, index_start=0, emb_size=7):
-        random_batch = np.random.random([nr, emb_size]).astype(np.float32)
-        for i in range(index_start, nr + index_start):
+        for i in range(index_start, index_start + nr):
```
Comment (cristianmtr) [inline tests/conftest.py]: Why did you change this? Please revert it, as it was optimized for the benchmarking tests

File: tests/conftest.py [hunk 2]
```diff
@@ -23,16 +23,15 @@ def docker_compose(request):
...
+            d.embedding = np.random.random(emb_size).astype(np.float32)
             yield d
 
     return get_documents_inner
 
 
 @pytest.fixture()
 def runtime_args():
-    return {'pea_id': 0, 'replica_id': 0, 'parallel': 1}
+    return {'shard_id': 0, 'replica_id': 0, 'shards': 1}
```
Comment (cristianmtr) [inline tests/conftest.py:38]: Is `parallel` gone completely?

**Review by cristianmtr — CHANGES_REQUESTED** [review]

File: tests/conftest.py (hunk 2 above)
Comment (JoanFM) [reply tests/conftest.py:38]: yes

File: tests/conftest.py (hunk 1 above)
Comment (JoanFM) [reply tests/conftest.py]: please rereview

**Review by cristianmtr — APPROVED** [review]

File: tests/conftest.py [hunk 3]
```diff
@@ -23,7 +23,7 @@ def docker_compose(request):
 @pytest.fixture()
 def get_documents():
     def get_documents_inner(nr=10, index_start=0, emb_size=7):
-        random_batch = np.random.random([nr, emb_size]).astype(np.float32)
+        random_batch = np.random.random(nr, emb_size).astype(np.float32)
```
Comment (JoanFM) [inline tests/conftest.py]:
```suggestion
        random_batch = np.random.random([nr, emb_size]).astype(np.float32)
```

**Review by cristianmtr — APPROVED** [review]
